[add] Open Library & Library Work Flow pages

dethan3 opened the pull request:
[![PR-20](https://badgen.net/badge/GitHub.dev/PR-20/blue?icon=visualstudio)](https://github.dev/Open-Source-Bazaar/Open-Source-Bazaar.github.io/pull/20) [![PR-20](https://badgen.net/badge/GitHub%20codespaces/PR-20/black?icon=github)](https://codespaces.new/Open-Source-Bazaar/Open-Source-Bazaar.github.io/pull/20) [![PR-20](https://badgen.net/badge/GitPod.io/PR-20/orange?icon=git)](https://gitpod.io/?autostart=true#https://github.com/Open-Source-Bazaar/Open-Source-Bazaar.github.io/pull/20) [<img width="16" alt="Powered by Pull Request Badge" src="https://user-images.githubusercontent.com/1393946/111216524-d2bb8e00-85d4-11eb-821b-ed4c00989c02.png">](https://pullrequestbadge.com/?utm_medium=github&utm_source=Open-Source-Bazaar&utm_campaign=badge_info)<!-- PR-BADGE: PLEASE DO NOT REMOVE THIS COMMENT -->  

Checklist（清单）:

<!-- Please follow this checklist and put an x in each of the boxes, like this: [x].（请遵循此清单，并在每个 [ ] 中输入 x，如下所示：[x]。） -->

- [x] Labels
- [x] Assignees
- [x] Reviewers

1. Add [Open Library], 
2. Use the documentation to quickly start operating Open Library.

<!--If your pull request closes a GitHub issue, replace the XXXXX below with the issue number.（如果 pull request 关闭一个 GitHub issue，请用 GitHub issue 编号替换下面的 XXXXX）-->

Closes #15

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

* **New Features**
  * 新增 Open Library 专区：主页、图书目录页、图书详情页与“如何借阅”指南页。
  * 新增专区组件：精选图书、图书卡片、步骤/规则展示、页头与页脚，并提供“查看详情/查看全部/借阅申请”入口。
  * 新增图书数据模型与 Open Library 图书目录 API，为页面提供静态数据来源。
* **Bug Fixes**
  * 图书详情改为按路由服务端加载；不存在会正确返回未找到状态。
* **Refactor**
  * 调整站点框架：Open Library 与通用站点布局按路由分流渲染页脚与导航。
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

dethan3 committed:
library flow

dethan3 committed:
[feat] Initial open-library homepage, component & hook fix

dethan3 committed:
feat: customize Open Library layout and styling

dethan3 committed:
fix: resolve import sorting and TS annotation issues

dethan3 committed:
feat: implement Open Library basic framework with multilingual support

dethan3 committed:
fix: replace <a> tags with Next.js <Link> components and fix TypeScript errors

dethan3 committed:
feat: Complete MVP layout improvements

dethan3 committed:
Merge upstream/main and resolve conflicts

TechQuery commented on components/open-library/Footer.tsx:
```diff
@@ -0,0 +1,116 @@
+                <i className="bi bi-wechat me-2" />
+                WeChat: FCCChengdu
+              </p>
+            </div>
```
```suggestion
            <ul className="list-unstyled d-flex flex-column gap-2 text-light opacity-75">
              <li>
                <Icon name="geo-alt" className="me-2" />
                freeCodeCamp Chengdu Community
              </li>
              <li>
                <i className="bi bi-pin-map me-2" />
                Chengdu, Sichuan, China
              </li>
              <li>
                <i className="bi bi-envelope me-2" />
                Email: <a href="mailto:team@fcc-cd.dev">team@fcc-cd.dev</a>
              </li>
              <li>
                <i className="bi bi-wechat me-2" />
                <a target="_blank" href="https://open.weixin.qq.com/qr/code?username=gh_b8b06d05cfa6">
                  WeChat: freeCodeCamp 成都社区
                </a>
              </li>
            </ul>
```
前两行的地址翻译一下

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+    );
+  }
+
+  if (!book) {
+    return (
+      <Layout title="Book Not Found - Open Library">
+        <ContentContainer>
+          <div className="text-center my-5">
+            <h2>Book Not Found</h2>
+            <p>Sorry, we couldn't find the book you're looking for.</p>
+            <Button
+              variant="primary"
+              onClick={() => router.push('/open-library/books')}
+            >
+              Return to Catalog
+            </Button>
+          </div>
+        </ContentContainer>
+      </Layout>
+    );
+  }
```
后端接口 404 后都不会执行本页面组件的代码，会统一在 `pages/_error.tsx` 处理。

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+  const [book, setBook] = useState<Book | null>(null);
+  const [loading, setLoading] = useState(true);
+
+  useEffect(() => {
+    if (id) {
+      // In a real app, this would be an API call
+      const bookId = Array.isArray(id)
+        ? parseInt(id[0])
+        : parseInt(id as string);
+      const foundBook = booksDatabase.find(b => b.id === bookId);
+
+      setBook(foundBook || null);
+      setLoading(false);
+    }
+  }, [id]);
+
+  if (loading) {
+    return (
+      <Layout title="Loading... - Open Library">
+        <ContentContainer>
+          <div
+            className="d-flex justify-content-center align-items-center"
+            style={{ height: '50vh' }}
+          >
+            <div className="spinner-border text-primary" role="status">
+              <span className="visually-hidden">Loading...</span>
+            </div>
+          </div>
+        </ContentContainer>
+      </Layout>
+    );
+  }
```
服务端渲染不会在后端数据加载完成前渲染任何 DOM，此处只有纯客户端渲染有用。

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+                </Col>
+                <Col md={9}>
+                  <h1>{book.title}</h1>
+                  <h5 className="text-muted mb-3">by {book.author}</h5>
```
作者信息不是标题，一般用 `<cite />`：https://developer.mozilla.org/en-US/docs/Web/HTML/Reference/Elements/cite

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+
+                  <p className="lead">{book.description}</p>
+
+                  <Row className="mt-4">
+                    <Col sm={6} md={3} className="mb-3">
+                      <div className="text-muted small">ISBN</div>
+                      <div>{book.isbn || 'N/A'}</div>
+                    </Col>
+                    <Col sm={6} md={3} className="mb-3">
+                      <div className="text-muted small">Publisher</div>
+                      <div>{book.publisher || 'N/A'}</div>
+                    </Col>
+                    <Col sm={6} md={3} className="mb-3">
+                      <div className="text-muted small">Published</div>
+                      <div>{book.publishYear || 'N/A'}</div>
+                    </Col>
+                    <Col sm={6} md={3} className="mb-3">
+                      <div className="text-muted small">Pages</div>
+                      <div>{book.pageCount || 'N/A'}</div>
+                    </Col>
+                  </Row>
```
```suggestion
                  <Row as="ul" className="list-unstyled mt-4 g-3" sm={2} md={4}>
                    <Col as="li">
                      <div className="text-muted small">ISBN</div>
                      <div>{book.isbn || 'N/A'}</div>
                    </Col>
                    <Col as="li">
                      <div className="text-muted small">Publisher</div>
                      <div>{book.publisher || 'N/A'}</div>
                    </Col>
                    <Col as="li">
                      <div className="text-muted small">Published</div>
                      <div>{book.publishYear || 'N/A'}</div>
                    </Col>
                    <Col as="li">
                      <div className="text-muted small">Pages</div>
                      <div>{book.pageCount || 'N/A'}</div>
                    </Col>
                  </Row>
```

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+          <div className="mt-4">
+            <Tabs defaultActiveKey="reviews" className="mb-3">
+              <Tab eventKey="reviews" title="Reviews">
+                {book.reviews && book.reviews.length > 0 ? (
+                  <div>
```
```suggestion
                {book.reviews?.[0] ? (
                  <>
```

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+              </Tab>
+              <Tab eventKey="history" title="Borrow History">
+                {book.borrowHistory && book.borrowHistory.length > 0 ? (
+                  <div className="table-responsive">
+                    <table className="table table-hover">
```
用 `<Table />`：https://react-bootstrap.github.io/docs/components/table/

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,456 @@
+                          <tr key={index}>
+                            <td>{history.borrower}</td>
+                            <td>
+                              {new Date(
+                                history.borrowDate,
+                              ).toLocaleDateString()}
```
```suggestion
                              {formatDate(history.borrowDate, 'YYYY-MM-DD')}
```
1. 服务端地区设置和用户浏览器通常不同，`toLocaleXXX()` 的结果也会不同，会出现 SSR 报错
2. https://web-cell.dev/web-utility/functions/formatDate.html

TechQuery reviewed (changes requested):
1. 很多问题在不同文件都有，不一一列举，但都要改
2. 软件开发要一个一个模块来，代码量太大对开发者、reviewer、AI 都有巨大上下文负担，很容易跑偏、卡死
3. 以后要在看板上多拆几个 issue，不要闷头搞很久后发一个差不多做完整个项目的巨大 PR……

dethan3 committed:
Merge remote-tracking branch 'upstream/main'

dethan3 commented on package.json:
```diff
@@ -93,5 +93,6 @@
     "*.{html,md,scss,json,yml,js,mjs,ts,tsx}": "prettier --write"
-  }
+  },
+  "packageManager": "pnpm@10.12.4+sha512.5ea8b0deed94ed68691c9bad4c955492705c5eeb8a87ef86bc62c74a26b037b08ff9570f108b2e4dbd1dd1a9186fea925e527f141c648e85af45631074680184"
```
这里是 Corepack 自动添加的，每次提交前我都会手动删除，上次忘了，以后注意

TechQuery replied:
> 这里是 Corepack 自动添加的，每次提交前我都会手动删除，上次忘了，以后注意

用的具体啥命令？

TechQuery commented on pages/open-library/how-to-borrow.tsx:
```diff
@@ -0,0 +1,298 @@
+            </Card.Body>
+          </Card>
+
+          <div className="text-center">
+            <h3 className="mb-4">准备好借阅了吗？</h3>
+            <div className="d-flex justify-content-center gap-3 flex-wrap">
+              <Link href="/open-library/books" legacyBehavior>
+                <a className="btn btn-primary btn-lg">浏览书籍目录</a>
+              </Link>
+              <a
+                href="https://open-source-bazaar.feishu.cn/share/base/form/shrcngQgMrhjTh6ycO1zcaEWZld"
+                target="_blank"
+                rel="noopener noreferrer"
+                className="btn btn-warning btn-lg"
+              >
+                申请成为会员
+              </a>
+            </div>
+          </div>
```
@dethan3 如果界面布局是经过特别设计的，就可以忽略这一条。

dethan3 committed:
Merge remote-tracking branch 'upstream/main'

# Conflicts:
#	.husky/pre-commit
#	.husky/pre-push
#	package.json
#	pages/_app.tsx
#	translation/en-US.ts
#	translation/zh-CN.ts
#	translation/zh-TW.ts

dethan3 committed:
refactor(open-library): address PR #20 review — i18n, types, mixin fixes

i18n / hardcoded strings:
- BookCard: replace hardcoded status text and cover alt with t()
- FeaturedBooks: add observer+useContext, replace hardcoded Chinese defaults with t()
- HowItWorks: add observer+useContext, move defaultSteps inside component, replace all hardcoded strings with t()
- Footer: fully i18n all text, replace placeholder hrefs with real URLs
- HeroSection: make all text props required, eliminate hardcoded defaults
- Navbar: replace hardcoded fallback with t('open_library')
- books/index.tsx: replace hardcoded subtitle with t()
- how-to-borrow.tsx: add observer+useContext, replace all section headings and buttons with t()
- index.tsx: replace all hardcoded strings in About/Support sections with t()
- book/[id].tsx: replace Publisher/Published/Pages/N/A labels with t()

translation keys: add ~20 new keys across zh-CN, en-US, zh-TW
(book_cover, learn_more_details, book_catalog_description, ready_to_borrow,
open_library_community_intro, our_mission_text, core_concepts, knowledge_flow,
knowledge_flow_desc, community_driven, community_driven_desc, open_sharing,
open_sharing_desc, support_us, support_us_description, share_your_book,
book_publisher, book_published_year, book_page_count, not_available)

type / style fixes:
- CardPage.tsx: replace Record<string, unknown> with ComponentType<any> to fix
  pre-existing TS2322 errors where FC<ActivityCardProps|Project|OrganizationCardProps>
  was not assignable to FC<Record<string,unknown>> under strict mode
- theme.less: fill empty semicolons in .section-frame() with correct mixin
  calls; add missing .button-base() to .button-primary() and .button-ghost()

TechQuery commented on styles/open-library.module.scss:
```diff
@@ -0,0 +1,11 @@
+  width: 100%;
+  max-width: 100%;
+  overflow-x: hidden;
+}
```
皆可用工具类声明

TechQuery commented on translation/zh-TW.ts:
```diff
@@ -456,8 +489,11 @@ export default {
+  community_location: '中國四川成都',
+  contact_email: 'team@fcc-cd.dev',
+  contact_address: '中國四川成都',
+  contact_wechat: 'freeCodeCamp 成都社群',
```
1. 重复的翻译文本写一次就行了
2. 邮箱直接硬编码，无需写在翻译文件

TechQuery commented on pnpm-workspace.yaml:
照抄 https://github.com/idea2app/Lark-Next-Bootstrap-ts/blob/c274370181050567a79fedc3ebeaad600ec6caa1/pnpm-workspace.yaml 后执行 `pnpm install`。

TechQuery commented on pages/open-library/index.tsx:
```diff
@@ -162,73 +98,50 @@ export default function OpenLibraryHomepage() {
-                    分享你的图书
-                  </span>
+                  ❤️ GitHub Sponsors
+                </Button>
```
我们目前没有开通赞助渠道，这个链接也不对，放在这里是干嘛的？

TechQuery commented on components/Layout/CardPage.tsx:
```diff
@@ -1,10 +1,11 @@
 export interface CardPageProps extends SearchPageMeta {
-  Card: ComponentClass<any> | FC<any>;
+  // eslint-disable-next-line @typescript-eslint/no-explicit-any
+  Card: ComponentType<any>;
```
可以加注释，但类型不要改啊！

TechQuery commented on components/open-library/Layout.tsx:
这么简单的结构直接在每个页面上用 `<Container />` 组件就行了，不用再封装一层了。

TechQuery commented on pages/_app.tsx:
```diff
@@ -38,6 +40,68 @@ export default class CustomApp extends App<I18nProps> {
   }
 
+  renderFooter(thisFullYear: number, t: (key: 'open_source_bazaar') => string) {
+    return (
```
```suggestion
  renderFooter(thisFullYear: number) {
    const { t } = this.i18nStore;

    return (
```

TechQuery commented on pages/_app.tsx:
```diff
@@ -38,6 +40,68 @@ export default class CustomApp extends App<I18nProps> {
+        <div className="mt-5 pt-2">{isArticlePage ? <PageContent>{content}</PageContent> : content}</div>
+      </>
+    );
+  }
```
```suggestion
  renderOpenLibraryFrame() {
    const { Component, pageProps } = this.props;

    return (
      <>
        <LibraryNavbar />
        <main className="py-5"><Component {...pageProps} /></main>
        <FooterComponent />
      </>
    );
  }

  renderSiteFrame(isArticlePage: boolean) {
    const { Component, pageProps } = this.props;
    const content = <Component {...pageProps} />;

    return (
      <>
        <MainNavigator />

        <div className="mt-5 pt-2">{isArticlePage ? <PageContent>{content}</PageContent> : content}</div>
      </>
    );
  }
```
现在这 AI 都咋训练的？明明是面向对象代码，却用着函数式编程的思路…… 灌了多少 React hooks 的语料？

TechQuery commented on pages/_app.tsx:
```diff
@@ -54,41 +121,13 @@ export default class CustomApp extends App<I18nProps> {
-          </p>
-        </footer>
+        {isActivityPage
+          ? this.renderPageContent(Component, pageProps)
```
```suggestion
          ? <Component {...pageProps} />
```

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,225 @@
+  return { props: { book } };
+};
+
+const BookDetail = observer(({ book }: BookDetailProps) => {
```
```suggestion
export const getServerSideProps: GetServerSideProps<Book> = async ({ params }) => {
  const bookId = +(params!.id + '');
  const book = openLibraryBooks.find(({ id }) => id === bookId);

  return book ? { props: book } : { notFound: true };
};

const BookDetail: FC<BookDetailProps> = observer(({ title, cover }) => {
```
运行时这个页面能进入，有且只有一个 ID。

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,225 @@
+          </Button>
+
+          <Card className="border-0 shadow-sm">
+            <Card.Body>
```
```suggestion
          <Card className="border-0 shadow-sm" body>
```

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,225 @@
+                          </div>
+                          <Card.Text>{review.comment}</Card.Text>
+                          <Card.Text>
+                            <small className="text-muted">
```
```suggestion
                      <Card key={index} className="mb-3 shadow-sm" body>
                          <div className="d-flex justify-content-between align-items-center mb-2">
                            <h5 className="mb-0">{review.reviewer}</h5>
                            <ol className="list-unstyled">
                              {[...Array(5)].map((_, i) => (
                                <li key={i} className="text-warning">
                                  {i < review.rating ? '\u2605' : '\u2606'}
                                </li>
                              ))}
                            </ol>
                          </div>
                          <Card.Text>{review.comment}</Card.Text>
                          <Card.Text className="small text-muted">
```

TechQuery commented on pages/open-library/how-to-borrow.tsx:
```diff
@@ -0,0 +1,235 @@
+        中查找自己感兴趣的书籍，或者在我们的 <Link href="/open-library/books">书籍目录</Link>{' '}
+        中浏览。
+      </>
+    ),
```
用函数式 i18n 翻译键来拼接外部传入的链接字符串，链接、加粗等文本格式的最终渲染可用 `marked()` 转换。

dethan3 committed:
Address Open Library review feedback

dethan3 committed:
Fix Open Library review feedback

dethan3 commented on pages/_app.tsx:
```diff
@@ -38,6 +40,60 @@ export default class CustomApp extends App<I18nProps> {
+            target="_blank"
+            rel="noopener noreferrer"
+          >
+            Powered by
```
这里不做修改

dethan3 commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,210 @@
+                style={{ maxHeight: '300px' }}
+                width={200}
+                height={300}
+              />
```
你的判断判断不完全准确。Next 文档说明 src 以 .svg 结尾时会自动走 unoptimized，不应直接启用 dangerouslyAllowSVG ，这里暂时不做修改

dethan3 committed:
Reject protocol-relative Open Library links

TechQuery commented on components/open-library/FeaturedBooks.tsx:
```diff
@@ -0,0 +1,78 @@
+  },
+);
+
+export default FeaturedBooks;
```
不做异步加载就不需要默认导出，直接在声明处具名导出即可。

TechQuery commented on components/open-library/HeroSection.tsx:
```diff
@@ -0,0 +1,73 @@
+  heroImageAlt: string;
+}
+
+const HeroSection: React.FC<HeroSectionProps> = ({
```
```suggestion
import { FC } from 'react';
import { Button, Col, Container, Image, Row } from 'react-bootstrap';

export interface HeroSectionProps {
  title: string;
  subtitle: string;
  ctaText: string;
  ctaLink: string;
  browseText: string;
  browseLink?: string;
  heroImage: string;
  heroImageAlt: string;
}

export const HeroSection: FC<HeroSectionProps> = ({
```
全局各处优先用具名的导入导出。

TechQuery commented on components/open-library/HowItWorks.tsx:
```diff
@@ -0,0 +1,126 @@
+          </div>
+
+          <Row as="ol" className="g-4 list-unstyled mb-0">
+            {resolvedSteps.map((step, index) => (
```
对象参数解构

TechQuery commented on pages/_app.tsx:
```diff
@@ -46,6 +102,8 @@ export default class CustomApp extends App<I18nProps> {
     const isArticlePage = asPath.startsWith('/article/') || asPath.startsWith('/policy/'),
       isActivityPage = asPath.startsWith('/hackathon');
 
+    const isOpenLibraryPath = asPath.startsWith('/open-library');
```
```suggestion
       isActivityPage = asPath.startsWith('/hackathon'),
       isOpenLibraryPath = asPath.startsWith('/open-library');
```

TechQuery commented on pages/_app.tsx:
```diff
@@ -56,39 +114,13 @@ export default class CustomApp extends App<I18nProps> {
-          </p>
-        </footer>
+
+        {!isActivityPage && !isOpenLibraryPath && this.renderFooter(thisFullYear)}
```
把此处逻辑合并到上面三目运算符的最后一个分支。

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,210 @@
+  return book ? { props: book } : { notFound: true };
+};
+
+const BookDetail: FC<Book> = observer(book => {
```
对象参数解构

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,210 @@
+                className="img-fluid"
+                style={{ maxHeight: '300px' }}
+                width={200}
+                height={300}
```
```suggestion
                style={{ maxHeight: '300px' }}
                width={200}
```

dethan3 replied:
https://github.com/Open-Source-Bazaar/Open-Source-Bazaar.github.io/pull/20#discussion_r3522370566
按照建议代码修改，不符合 Next.js 的规范，请问应该如何选择？

TechQuery replied:
> https://github.com/Open-Source-Bazaar/Open-Source-Bazaar.github.io/pull/20#discussion_r3522370566
> 按照建议代码修改，不符合 Next.js 的规范，请问应该如何选择？

Next.js 各种规范真是奇怪，先按它说的来吧……

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,210 @@
+                        {formatDate(review.date, 'YYYY-MM-DD')}
+                      </Card.Text>
+                    </Card>
+                  ))}
```
这么深的回调函数嵌套，要么把它抽成独立函数组件，要么把整个大组件写成类组件，再把这个渲染回调写成类的实例方法。（下同）

TechQuery commented on pages/open-library/how-to-borrow.tsx:
```diff
@@ -0,0 +1,255 @@
+  if (cursor < text.length) nodes.push(text.slice(cursor));
+
+  return nodes;
+};
```
你要操作的这些文本都存在本仓库管理的代码中，不是外部用户输入，都是安全、可控的，直接用 `marked()` 解析后去 setInnerHTML 就行了，不用搞这么麻烦！

TechQuery replied:
你要操作的这些文本都存在本仓库管理的代码中，不是外部用户输入，都是安全、可控的，直接用 `marked()` 解析后去 setInnerHTML 就行了，不用搞这么麻烦！

TechQuery replied:
你要操作的这些文本都存在本仓库管理的代码中，不是外部用户输入，都是安全、可控的，直接用 `marked()` 解析后去 setInnerHTML 就行了，不用搞这么麻烦！

TechQuery commented on pages/open-library/how-to-borrow.tsx:
```diff
@@ -0,0 +1,255 @@
+                    >
+                      {t('view_full_catalog')}
+                    </Button>
+                  </ListGroup.Item>
```
```suggestion
                <ListGroup className="list-unstyled d-grid gap-2 mb-0" variant="flush" aria-label={t('open_library_quick_links_label')}>
                  <ListGroup.Item
                      href={catalogURL}
                      target="_blank"
                      rel="noopener noreferrer"
                    >
                      {t('view_full_catalog')}
                  </ListGroup.Item>
```
这个组件本身就有导航链接模式，不需要自己嵌套：https://react-bootstrap.github.io/docs/components/list-group/#actionable-items

TechQuery commented on translation/en-US.ts:
```diff
@@ -339,4 +339,245 @@ export default {
+  }: {
+    catalogURL: string;
+    booksURL: string;
+  }) =>
```
`Record`

TechQuery reviewed (changes requested):
所有类似问题都要全局改

dethan3 committed:
Address Open Library review feedback

TechQuery commented on components/open-library/BookCard.tsx:
```diff
@@ -0,0 +1,75 @@
+                bg={book.status === 'available' ? 'success' : 'warning'}
+                text={book.status === 'available' ? undefined : 'dark'}
+              >
+                {book.status === 'available' ? t('available') : t('borrowed')}
```
多次使用的表达式抽出一个常量

TechQuery commented on components/open-library/BookCard.tsx:
```diff
@@ -0,0 +1,75 @@
+                height: '180px',
+                objectFit: 'contain',
+                padding: '10px',
+              }}
```
```suggestion
              className="rounded bg-light object-fit-contain"
              style={{ height: '180px', padding: '10px' }}
```

TechQuery commented on components/open-library/Footer.tsx:
```diff
@@ -0,0 +1,89 @@
+                💬{' '}
+                <a
+                  className="text-light"
+                  href="https://open.weixin.qq.com/qr/code?username=gh_b8b06d05cfa6"
```
多次出现的重复网址也要像上面的邮箱一样声明一个常量，也可以集中放到专门的 `model/configuration.ts`。

TechQuery commented on components/open-library/HowItWorks.tsx:
```diff
@@ -0,0 +1,124 @@
+  description: string;
+  icon: string;
+  color: string;
+}
```
把 `string` 类型的字段都放到 `extends` 后面的 `Record`。

TechQuery commented on components/open-library/Navbar.tsx:
```diff
@@ -0,0 +1,30 @@
+        <Navbar.Toggle aria-controls="basic-navbar-nav" />
+        <Navbar.Collapse id="basic-navbar-nav">
+          <Nav className="me-auto">
+            {/* <Nav.Link href="/open-library">{t('home')}</Nav.Link> */}
```
```suggestion
```

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,267 @@
+      </div>
+    );
+
+  return (
```
用三目运算符合起来

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,267 @@
+          reviewer={reviewer}
+          rating={rating}
+          comment={comment}
+          date={date}
```
用展开简写的对象自变量

TechQuery reviewed (changes requested):
@copilot 所有的问题全篇都要改类似的情况

dethan3 committed:
Address Open Library review follow-ups

TechQuery commented on components/Footer.tsx:
这个页脚组件的设计可以直接用在全局，把老的页脚删掉就行了，文件夹也提升一层。

TechQuery commented on components/open-library/HeroSection.tsx:
```diff
@@ -0,0 +1,68 @@
+      'title' | 'subtitle' | 'ctaText' | 'ctaLink' | 'browseText' | 'heroImage' | 'heroImageAlt',
+      string
+    >,
+    Partial<Record<'browseLink', string>> {}
```
我之前的建议核心目的在于“消除重复”，这个单项的它不重复，你这么写反而会让代码更长，`interface` 的 body 却变成了一个空的花括号……

TechQuery commented on models/configuration.ts:
```diff
@@ -30,6 +30,23 @@ export const LARK_API_HOST = `${API_Host}/api/Lark/`;
+
+export const OpenLibraryGitHubURL =
+  'https://github.com/Open-Source-Bazaar/Open-Source-Bazaar.github.io';
+export const OpenLibraryWeChatURL = 'https://open.weixin.qq.com/qr/code?username=gh_b8b06d05cfa6';
```
这些是整个项目的通用信息，不应该加 `OpenLibrary` 前缀。

TechQuery commented on models/configuration.ts:
```diff
@@ -30,6 +30,23 @@ export const LARK_API_HOST = `${API_Host}/api/Lark/`;
+export const OpenLibraryMembershipFormURL =
+  'https://open-source-bazaar.feishu.cn/share/base/form/shrcngQgMrhjTh6ycO1zcaEWZld';
+export const OpenLibraryReviewFormURL =
+  'https://open-source-bazaar.feishu.cn/share/base/form/shrcnFAX8q7mnywCQOsVnIS5Dwb';
```
具体某个子系统的环境变量放到这个文件的最下方，比较通用的才放在上面。

TechQuery commented on pages/open-library/book/[id].tsx:
```diff
@@ -0,0 +1,257 @@
+        const review = { reviewer, rating, comment, date };
+
+        return <ReviewCard key={`${reviewer}-${date}`} {...review} />;
+      })}
```
```suggestion
      {reviews.map(({ reviewer, rating, comment, date }) =>
        <ReviewCard key={`${reviewer}-${date}`} {...{ reviewer, rating, comment, date }} />
      )}
```

TechQuery commented on pages/open-library/how-to-borrow.tsx:
```diff
@@ -0,0 +1,199 @@
+
+const booksURL = '/open-library/books';
+
+const trustedInlineMarkupOf = (text: string) => ({ __html: marked.parseInline(text) as string });
```
```suggestion
const trustedInlineMarkupOf = (text: string) => ({ __html: marked(text) as string });
```
直接把这个函数返回的东西写在每一处调用代码即可，内容太少，无需封装。

TechQuery reviewed (changes requested):
全局所有类似问题都要改。

如果是 AI 智能体在写代码的话，它需要领会每一处建议的用意，而非盲目地从一个极端走向另一个极端，需要体现出智能。

dethan3 committed:
Address remaining PR review feedback

dethan3 committed:
Add hackathon to spellcheck dictionary

TechQuery commented on components/open-library/HeroSection.tsx:
```diff
@@ -0,0 +1,67 @@
+  extends
+    Record<'title' | 'subtitle' | 'ctaText' | 'ctaLink' | 'browseText' | 'heroImage', string>,
+    Partial<Record<'browseLink', string>> {
+  heroImageAlt: string;
```
```suggestion
    Record<'title' | 'subtitle' | 'ctaText' | 'ctaLink' | 'browseText' | 'heroImage' | 'heroImageAlt', string> {
  browseLink?: string;
```

TechQuery commented on pages/open-library/how-to-borrow.tsx:
```diff
@@ -0,0 +1,199 @@
+                    <div>
+                      <h4 className="h5">{title}</h4>
+                      <p
+                        dangerouslySetInnerHTML={{ __html: marked.parseInline(description) }}
```
```suggestion
                        dangerouslySetInnerHTML={{ __html: marked(description) as string }}
```

TechQuery commented on pages/_app.tsx:
```diff
@@ -38,13 +39,44 @@ export default class CustomApp extends App<I18nProps> {
+  renderOpenLibraryFrame() {
+    const { Component, pageProps } = this.props;
+
+    return (
+      <>
+        <LibraryNavbar />
+        <main className="py-5">
+          <Component {...pageProps} />
+        </main>
+        <Footer />
+      </>
+    );
+  }
+
+  renderSiteFrame(isArticlePage: boolean) {
+    const { Component, pageProps } = this.props;
+    const content = <Component {...pageProps} />;
+
+    return (
+      <>
+        <MainNavigator />
+
+        <div className="mt-5 pt-2">
+          {isArticlePage ? <PageContent>{content}</PageContent> : content}
+        </div>
+
+        <Footer />
+      </>
+    );
+  }
```
`<Footer />` 是全局的，还是放到主分支原来的位置。

TechQuery reviewed (changes requested):
@dethan3 你用的是哪家的 AI IDE？怎么看不懂人话呢？经常做一半…… 还有切记要处理冲突！

dethan3 committed:
Merge upstream/main and address review feedback
